iothub/device/src/InternalClient.cs

```diff
@@ -156,6 +157,8 @@ public InternalClient(IotHubConnectionString iotHubConnectionString, ITransportS
                 Logging.Enter(this, transportSettings, pipelineBuilder, nameof(InternalClient) + "_ctor");
             }
 
+            JsonConvert.DefaultSettings = () => new JsonSerializerSettings { MaxDepth = 128 };
```

Comment:
You'll want to add some tests that validate that this value can be overridden by the user.

Reply:
I added a few test cases. Could you take a look at them? Thank you in advance!